### ISSUE TITLE
Needs complete refactoring. My two cents. 

### ISSUE THREAD

**Issue by timothymarois** [BingAds/BingAds-PHP-SDK#68]
Obviously, there are no PHP devs at Microsoft. **This is one terrible API SDK**. I mean no offense to the author,  but hire a PHP developer. There are more problems in this API than I can even list right now. The biggest thing that is incredibly irritating is the use of `$GLOBALS` are we coding in 1999 again? The entire sample package is utterly useless. You would have to rewrite the whole damn thing to get it integrated into any modern platform. Now you'll come back and say. "That's why it's a sample." But its worse than just looking at examples, because all of your methods require dependencies across the board, its a spider-web, a bunch of spaghetti code. This API breaks all standard coding practices.

**Comment by J7mbo** [comment]
Hey! A few minor points before the response that you may not be aware of - it’s an SDK not an API, and a tool auto-generates the code, they don’t write it themselves. So they make changes to this tool that autogenerates it.

Regardless, it doesn’t matter. Even if this library was perfect, you’d do well to protect yourself with an anti-corruption layer *anyway*. So do that.

As for globals, I haven’t used that once.

**Comment by timothymarois** [comment]
@J7mbo I wasn't aware a machine wrote this code. That might make a lot more sense now.  I wonder if we all out of a job soon? haha. Well anyways, thanks for the explanation.

**Comment by madsem** [comment]
I'm having the same concerns, and a hard time integrating the SDK.

I think this should stay open, so that @eric-urban and team take notice, and maybe change some things. A SDK is made for developers, so it should be easy to use and examples should be based on real-life situations and code.

The documentation is really messy, you have to jump around like a maniac to find things, examples are infested with $GLOBAL's, it's hard to follow the code even in the examples. Look at the Google Ads API... Much better documentation and their examples are based on real-life applications.

Please remember, not everyone who is using this API is a super-duper pro programmer or has backup from an entire department of other programmers, there are many who are using this API who automate their own small business.

**Comment by timothymarois** [comment]
Hey @madsem I decided to write my own wrapper around this [SDK for Laravel](https://github.com/tmarois/laravel-ads-sdk). My goal has been to make Google Ads and Bind Ads both function the same way with authentication, reports and account management. Its not entirely complete, but it does the job for the most part.
